### PR TITLE
Electric Chair Refactor

### PR DESCRIPTION
**Pull request by TealSeer**
[REWORK] [GAME OBJECTS]
## About the PR 

- Move settings for electric chair from HTML menu accessed by right-click verb to context menu accessed by multitool (sprites by Joeled)
- Removes SPAWN in New() and changes proc signature to take constitute parts for assembly as args instead of waiting for caller to assign them
- Adds help message for accessing settings menu
- Adds option to configure the signal/code of the electric chair after it's been assembled
- Fix bug where electric chairs could be rotated and would break their sprite when rotated north
- Replace 0/1 usage in electric chair code with FALSE/TRUE defines

image

## Why's this needed? 

Right-click verbs are the devil and so are HTML menus and so is `SPAWN`

## Changelog 

```changelog
(u) TealSeer & Joeled
(+) The electric chair has a new settings menu that can be accessed by clicking with a multitool.
```

### CODE REVIEW

**Review by Tarmunora — APPROVED** [review]

File: code/obj/item/assembly/shock_kit.dm [hunk 1]
```diff
@@ -7,16 +7,16 @@
 	w_class = W_CLASS_HUGE
 	flags = FPRINT | TABLEPASS| CONDUCT
 
-/obj/item/assembly/shock_kit/New()
+/obj/item/assembly/shock_kit/New(turf/newLoc, obj/item/clothing/head/helmet/helmet, obj/item/device/radio/electropack/electropack)
```
Comment (ZeWaka) [inline code/obj/item/assembly/shock_kit.dm:10]:
```suggestion
/obj/item/assembly/shock_kit/New(atom/newLoc, obj/item/clothing/head/helmet/helmet, obj/item/device/radio/electropack/electropack)
```
not a turf

File: code/obj/stool.dm [hunk 2]
```diff
@@ -1473,85 +1470,74 @@ TYPEINFO(/obj/stool/chair/dining/wood)
...
-	securable = 0
+	securable = FALSE
+	var/list/datum/contextAction/contexts = list()
+	HELP_MESSAGE_OVERRIDE("")
 
-	New()
+	get_help_message(dist, mob/user)
+		. = "You can use a <b>multitool</b> to open the settings menu, or a <b>wrench</b> to disassemble it."
+
+	New(turf/newLoc, obj/item/assembly/shock_kit/shock_kit)
```
Comment (ZeWaka) [inline code/obj/stool.dm:1496]:
```suggestion
	New(atom/newLoc, obj/item/assembly/shock_kit/shock_kit)
```
x

File: code/obj/stool.dm [hunk 3]
```diff
@@ -1473,85 +1470,74 @@ TYPEINFO(/obj/stool/chair/dining/wood)
 /* -------------------- Electric Chairs -------------------- */
 /* ========================================================= */
 
+#define ELECTRIC_CHAIR_TOGGLE_POWER 0
+#define ELECTRIC_CHAIR_TOGGLE_LETHAL 1
+#define ELECTRIC_CHAIR_SHOCK 2
+#define ELECTRIC_CHAIR_SET_SIGNAL 3
 /obj/stool/chair/e_chair
```
Comment (ZeWaka) [inline code/obj/stool.dm:1477]: Please just move this stuff to a different file

File: code/obj/item/assembly/shock_kit.dm [hunk 4]
```diff
@@ -7,16 +7,16 @@
...
-	SPAWN(2 SECONDS)
-		if (src)
-			if (!(src.part1 && istype(src.part1)))
-				src.part1 = new /obj/item/clothing/head/helmet(src)
-				src.part1.master = src
-			if (!(src.part2 && istype(src.part2)))
-				src.part2 = new /obj/item/device/radio/electropack(src)
-				src.part2.master = src
+	if(!helmet)
+		helmet = new /obj/item/clothing/head/helmet(src)
```
Comment (ZeWaka) [inline code/obj/item/assembly/shock_kit.dm:13]:
```suggestion
	helmet ||= new /obj/item/clothing/head/helmet(src)
```
use this for the rest also

**Review by ZeWaka — CHANGES_REQUESTED** [review]

**Pull request merged**